core/store/src/trie/split_state.rs

```diff
@@ -157,9 +151,10 @@ impl ShardTries {
         let mut new_state_roots = state_roots.clone();
         let mut store_update = StoreUpdate::new_with_tries(self.clone());
         for (shard_uid, changes) in changes_by_shard {
-            let trie = self.get_trie_for_shard(shard_uid);
-            // Here we assume that state_roots contains shard_uid, the caller of this method will guarantee that
-            let trie_changes = trie.update(&state_roots[&shard_uid], changes.into_iter())?;
+            // Here we assume that state_roots contains shard_uid, the caller of this method will guarantee that.
+            let trie_changes = self
+                .get_trie_for_shard(shard_uid, state_roots[&shard_uid])
+                .update(changes.into_iter())?;
```

Comment:
Unrelated to the PR, but it would be idiomatic for `fn update` to accept `IntoIterator` rather than `Iterator`.